trillian_admin_api.proto

```diff
@@ -0,0 +1,92 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "com.google.trillian.proto";
+option java_outer_classname = "TrillianApiProto";
+
+package trillian;
+
+import "trillian.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/empty.proto";
+
+// ListTrees request.
+// No filters or pagination options are provided.
+message ListTreesRequest {}
+
+// ListTrees response.
+// No pagination is provided, all trees the requester has access to are
+// returned.
+message ListTreesResponse {
+  // Trees matching the list request filters.
+  repeated Tree tree = 1;
+}
+
+// GetTree request.
+message GetTreeRequest {
+  // ID of the tree to retrieve.
+  int64 tree_id = 1;
+}
+
+// CreateTree request.
+message CreateTreeRequest {
+  // Tree to be created. See Tree and CreateTree for more details.
+  Tree tree = 1;
+}
+
+// UpdateTree request.
+message UpdateTreeRequest {
+  // Tree to be updated.
+  Tree tree = 1;
+
+  // Fields modified by the update request.
+  // For example: "tree_state", "display_name", "description".
+  google.protobuf.FieldMask update_mask = 2;
+}
+
+// DeleteTree request.
+message DeleteTreeRequest {
+  // ID of the tree to delete.
+  int64 tree_id = 1;
+}
+
+// Trillian Administrative interface.
+// Allows creation and management of Trillian trees (both log and map trees).
+service TrillianAdmin {
```

Comment:
The service is indeed the most interesting part. I put it at the bottom because it depends on the messages above it, and someone told me once that that's how dependencies in proto files should be ordered. This seems to be consistent with the other proto files I've seen.